src/zope/interface/adapter.py

```diff
@@ -658,16 +658,21 @@ def _convert_None_to_Interface(x):
         return x
 
 def _lookup(components, specs, provided, name, i, l):
+    # this function is called very often.
+    # The components.get in loops is executed 100 of 1000s times.
+    # by loading get into a local variable the bytecode
+    # "LOAD_FAST 0 (components)" in the loop can be eliminated.
+    cget = components.get
     if i < l:
         for spec in specs[i].__sro__:
-            comps = components.get(spec)
+            comps = cget(spec)
```

Comment:
FWIW (which is not much), when I write this type of optimization, I tend to name the local variable the same as the full method call, only using an underscore:
```python
components_get = components.get
comps = components_get(spec)
```

That serves as self-documenting code about what exactly it is we're optimizing (to me at least), and minimizes the risk of clashing with some other variable. Plus it doesn't change line lengths 😄